src/carcockpit/gltf_viewer_widget.hpp

```diff
@@ -0,0 +1,113 @@
+/*
+carcockpit - Car cockpit example GUI project
+
+Copyright (C) 2024 Gagistech Oy <gagisechoy@gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+/* ================ LICENSE END ================ */
+
+#pragma once
+
+#include <ruis/res/texture_2d.hpp>
+#include <ruis/res/texture_cube.hpp>
+#include <ruis/updateable.hpp>
+#include <ruis/widgets/base/fraction_widget.hpp>
+#include <ruis/widgets/widget.hpp>
+
+// #include "../ruis/render/scene/shaders/shader_adv.hpp"
+// #include "../ruis/render/scene/shaders/shader_phong.hpp"
+// #include "../ruis/render/scene/shaders/shader_skybox.hpp"
+#include "../ruis/render/scene/scene.hpp"
+#include "../ruis/render/scene/scene_renderer.hxx"
+
+#include "shaders/shader_adv.hpp"
+#include "shaders/shader_phong.hpp"
+#include "shaders/shader_skybox.hpp"
+
+namespace carcockpit {
+
+constexpr ruis::vec3 default_camera_position_front{0, 1.5, 7};
+constexpr ruis::vec3 default_camera_position_top{0, 9, 0};
+constexpr float default_camera_transition_duration{0.1};
+
+class gltf_viewer_widget : public ruis::fraction_widget, public ruis::updateable, virtual public ruis::widget
+{
+	std::shared_ptr<ruis::render::scene> demoscene;
+	std::shared_ptr<ruis::render::scene_renderer> sc_renderer;
+	std::shared_ptr<ruis::render::camera> camrip;
+	// std::shared_ptr<ruis::res::texture_cube> texture_environment_cube;
+
+	ruis::vec3 camera_position_front{default_camera_position_front};
```

Comment:
`camera_position_front` is actually `camera_direction`? Rename accordingly, if so